Source/stores.cpp

```diff
@@ -2543,6 +2547,18 @@ void S_ConfirmEnter()
 			break;
 		}
 		StartStore(stextshold);
+		if (forceSelection) {
+			stextsel = stextlhold;
+			stextsval = stextvhold; //making the selection stay after buying an item
+			if (stextsval >= 1) {
+				// this trick is the best way to use existing mechanics and make sure the selection is still valid

```

Comment:
Let's not use the trick and ensure in-range selection directly?
Then we won't need the `if (!forceSelection)` SFX disabling either

Reply:
are you saying I have any clue what's going on? :P

Reply:
Seeing are you say what am I saying and see??